File: php/j20.arguments.php
#!/usr/bin/php
<?php
// acquire num of args
$arg_count   = $argc - 1;  // get num of arguments

// iterative loop to enumerate args
for ( $count = $arg_count; $count > 0; $count-- )
  echo "item $count: $argv[$count]\n";
?>

File: perl/j20.arguments.pl
#!/usr/bin/perl
# acquire num of args
$arg_count = $#ARGV + 1; # get num of arguments
$last      = $#ARGV;     # get index of last element
$first     = 0;          # set index of first element 

print "The arguments passed are (reverse order):\n";
# iterative loop to enumerate args
for ( my $count = $last; $count >= $first ; $count-- ) {
  printf "  item %d: %s\n", $count+1, $ARGV[$count];
}